systemtests/tests/bconsole-pam/python-bareos-unittest.py

```diff
@@ -0,0 +1,196 @@
+#!/usr/bin/env python
+
+# -*- coding: utf-8 -*-
+
+from __future__ import print_function
+import logging
+import os
+import unittest
+
+import bareos.bsock
+import bareos.exceptions
+from   bareos.bsock.protocolversions import ProtocolVersions
+
+
+class PythonBareosBase(unittest.TestCase):
+    director_address = 'localhost'
+    director_port = 9101
+    director_root_password = 'secret'
+    director_operator_username = 'admin'
+    director_operator_password = 'secret'
+    console_pam_username = u'PamConsole'
+    console_pam_password = u'secret'
+    client = 'bareos-fd'
+    #restorefile = '/usr/sbin/bconsole'
+    # path to store logging files
+    backup_directory = 'tmp/data/'
+    debug = False
+    logpath = '{}/PythonBareosTest.log'.format(os.getcwd())
+
+    def setUp(self):
+        # Configure the logger, for information about the timings set it to INFO
+        logging.basicConfig(
+            filename=self.logpath,
+            format='%(levelname)s %(module)s.%(funcName)s: %(message)s',
+            level=logging.INFO)
+        logger = logging.getLogger()
+        if self.debug:
+            logger.setLevel(logging.DEBUG)
+        logger.debug('setUp')
+
+    def tearDown(self):
+        logger = logging.getLogger()
+        logger.debug('tearDown\n\n\n')
+
+    def get_name_of_test(self):
+        return self.id().split('.', 1)[1]
+
+
+class PythonBareosPamLoginTest(PythonBareosBase):
+    '''
+    Requires Bareos Console Protocol >= 18.2.
+    '''
+
+    def test_pam_login(self):
+
+        pam_username = u'user1'
+        pam_password = u'user1'
+
+        #
+        # login as console_pam_username
+        #
+        bareos_password = bareos.bsock.Password(self.console_pam_password)
+        director = bareos.bsock.DirectorConsole(
+            address=self.director_address,
+            port=self.director_port,
+            name=self.console_pam_username,
+            password=bareos_password,
+            pam_username=pam_username,
+            pam_password=pam_password)
+
+        whoami = director.call('whoami').decode('utf-8')
+        self.assertEqual(pam_username, whoami.rstrip())
+
+
+    def test_pam_login_with_not_existing_username(self):
+        '''
+        Verify bareos.bsock.DirectorConsole raises an AuthenticationError exception.
+        '''
+        pam_username = 'nonexistinguser'
+        pam_password = 'secret'
+
+        bareos_password = bareos.bsock.Password(self.console_pam_password)
+        with self.assertRaises(bareos.exceptions.AuthenticationError):
+            director = bareos.bsock.DirectorConsole(
+                address=self.director_address,
+                port=self.director_port,
+                name=self.console_pam_username,
+                password=bareos_password,
+                pam_username=pam_username,
+                pam_password=pam_password)
+
+
+    def test_login_with_not_wrong_password(self):
```

Comment:
```suggestion
    def test_login_with_right_password(self):
```

Reply:
In fact, the method name has been wrong. I changed it to "test_login_with_wrong_password".